integrations/Iterable/browser.js

```diff
@@ -0,0 +1,211 @@
+        onOpenScreenReaderMessage: this.onOpenScreenReaderMessage || undefined,
+        onOpenNodeToTakeFocus: this.onOpenNodeToTakeFocus || undefined,
+        packageName: this.packageName || undefined,
+        rightOffset: this.rightOffset || undefined,
+        topOffset: this.topOffset || undefined,
+        bottomOffset: this.bottomOffset || undefined,
+        handleLinks: this.handleLinks || undefined,
+        closeButton: {
+            color: this.closeButtonColor || 'red',
+            size: this.closeButtonSize || '16px',
+            topOffset: this.closeButtonColorTopOffset || '4%',
+            sideOffset: this.closeButtonColorSideOffset || '4%',
+            iconPath: this.iconPath || undefined,
+            isRequiredToDismissMessage: this.isRequiredToDismissMessage || undefined,
+            position: this.closeButtonPosition || 'top-right'
+          }
+        }
+    getInAppMessagesPayload = removeUndefinedAndNullValues(getInAppMessagesPayload);
+
+    const { request } = window['@iterable/web-sdk'].getInAppMessages(
+        getInAppMessagesPayload,
+        { display: 'immediate' }
+        );
+    // fetchAppEvents is a class function now available throughout
+    // we will trigger getInAppMessages when event name matches from the ui mapping in config.
+    this.fetchAppEvents = request;
+  }
+
+  track(rudderElement) {
+    logger.debug("===In track Iterable===");
+    
+    const { message } = rudderElement;
+    const { event } = message;
+    const eventPayload = removeUndefinedAndNullValues(message.properties);
+    const userEmail = get(message, "context.traits.email");
+    const userId = get(message, "userId");
+    
+    if (!event) {
+        logger.error("Event name not present");
+        return;
+    }
+    let isCustom = true;
+    if (isNotEmpty(this.getInAppEventMapping)) {
+        const mappedEvents = this.getInAppEventMapping;
+        mappedEvents.forEach((e) => {
+            if(e.eventName === event) {
+                // send a track call for getinappMessages if option enabled in config
+                if (this.sendTrackForInapp) {
+                    window['@iterable/web-sdk'].track({ email: userEmail, userId, eventName: "Track getInAppMessages", dataFields: eventPayload })
+                    .then(logger.debug("Track a getinappMessages event."));
+                }
+                this.fetchAppEvents();
+                isCustom = false;
+            }
+        })
+    }
+    if (isNotEmpty(this.purchaseEventMapping)) {
+        // purchase events
+        const mappedEvents = this.purchaseEventMapping;
+        mappedEvents.forEach((e) => {
+            if(e.eventName === event) {
+                const purchaseEventPayload = formPurchaseEventPayload(message)
+                window['@iterable/web-sdk'].trackPurchase(
+                purchaseEventPayload,
+                )
+                isCustom = false;
+            }
+        })
+    }
+    if (isCustom) {
```

Comment:
rather than we using flags and `if` we can use `else if` 
 ```
 if inApp --> do inApp work
 
 if purchase --> purchase work
 else if custom --> do custom
 ```

Reply:
I introduced isCustom flag mostly for debugging purposes going ahead, have removed this.

Reply:
If we use else-if, it will not invoke whenever there is some mapping present for purchase events. Thereby, used if so that it essentially after going through the purchase mapped event names in UI.